parsl/executors/high_throughput/process_worker_pool.py

```diff
@@ -731,9 +732,30 @@ def worker(
         os.sched_setaffinity(0, my_cores)  # type: ignore[attr-defined, unused-ignore]
         logger.info("Set worker CPU affinity to {}".format(my_cores))
 
+    # If CUDA devices, find total number of devices to allow for MPS
+    # See: https://developer.nvidia.com/system-management-interface
+    nvidia_smi_cmd = "nvidia-smi -L > /dev/null && nvidia-smi -L | wc -l"
+    nvidia_smi_ret = subprocess.run(nvidia_smi_cmd, shell=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+    if nvidia_smi_ret.returncode == 0:
+        num_cuda_devices = int(nvidia_smi_ret.stdout.split()[0])
+    else:
+        num_cuda_devices = None
+
     # If desired, pin to accelerator
     if accelerator is not None:
-        os.environ["CUDA_VISIBLE_DEVICES"] = accelerator
+        try:
+            if num_cuda_devices is not None:
+                procs_per_cuda_device = pool_size // num_cuda_devices
+                partitioned_accelerator = str(int(accelerator) // procs_per_cuda_device)  # multiple workers will share a GPU
+                os.environ["CUDA_VISIBLE_DEVICES"] = partitioned_accelerator
+                logger.info(f'Pinned worker to partitioned cuda device: {partitioned_accelerator}')
+            else:
+                os.environ["CUDA_VISIBLE_DEVICES"] = accelerator
+        except (TypeError, ValueError, ZeroDivisionError):
+            os.environ["CUDA_VISIBLE_DEVICES"] = accelerator
+        except Exception as e:
+            print(f"Unrecognized exception in setting cuda affinity: {e}\n")
```

Comment:
Perhaps do this to get it to go into the relevant worker_* log file rather than stdout.

```suggestion
            logger.exception("Unrecognized exception in setting cuda affinity")
```

... although I would also be happy with this not being caught, propagating out the top and killing the worker on the principle of "when there is weird stuff, let's not pretend it didn't happen"...

Reply:
after more thought, i am more convinced that this second catch of all exceptions should not be here

see https://gist.github.com/benclifford/1ad76311d082657bb8baf23de0063d18#exceptions for some notes i made

Reply:
Okay, so do you want me to remove the second catch all together?

Reply:
yes